### ISSUE TITLE
Allow Verified, Sponsored, and Line extensions to appear in Sponsored shelf

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#15920]
Once we move to using Adzerk to determine what to show in the hero area we could use some additional flexibility. Instead of only allowing add-ons in the Sponsored group, it should be possible to have Verified or Line extensions there as well. This allows us to "pad" the group in case we don't have the full set of 6. 

The shelf will still be labeled Sponsored extensions, which may be slightly inaccurate if there are Line extensions there, but we can live with it.

**Comment by bobsilverberg** [comment]
I guess we're still going to call the endpoint `sponsored`, and just document that it can contain other categories as well?

**Comment by eviljeff** [comment]
> 
> 
> I guess we're still going to call the endpoint `sponsored`, and just document that it can contain other categories as well?

We could rename it, though I couldn't come up with a good alternative - partly because what it **is** keeps changing.  `/promoted/` would cover the different groups, though the output wouldn't include all promoted addons so is also inaccurate.  `/adzerk/` is the most accurate right now, but I don't want to tie the naming to the current backend service we're using to choose.

**Comment by jvillalobos** [comment]
If it helps with the naming, in the future we expect to also serve content for the primary hero area from Adzerk. So, if the same end point will be used for both, maybe something like "heroarea" or "homepagehero" would work. If they're different ones, it might be good to use names that differentiate the primary and secondary.

**Comment by bobsilverberg** [comment]
For now I think we'll continue to use separate endpoints for `sponsored` and `hero`, so I suggest we just keep all the names as is and possibly revisit them later when more functionality is added.

**Comment by eviljeff** [comment]
@jvillalobos now I re-read this, I'm not clear what's being requested - are you asking that if Verified and Line add-ons are added to Adzerk they will show up in the /sponsored endpoint shelf?  Or that if we get less than 6 sponsored add-ons from Adzerk then we should backfill the remainder with Verified/Line?

If the latter, what's the selection criteria - just a random selection from all Line/Verified add-ons?

**Comment by eviljeff** [comment]
Also, note that under the current design the fact that some other endpoint may also use Adzerk doesn't change anything about the possibility of less than 6 add-ons being returned - each api request is an Adzerk request and neither AMO nor Adzerk can de-duplicate them to avoid including the same add-on in two+ different places. 

Changing _that_ is a much bigger task (with frontend and api work) that needs the end requirements specified in a PRD, really

**Comment by bobsilverberg** [comment]
> Also, note that under the current design the fact that some other endpoint may also use Adzerk doesn't change anything about the possibility of less than 6 add-ons being returned - each api request is an Adzerk request and neither AMO nor Adzerk can de-duplicate them to avoid including the same add-on in two+ different places.
> 
> Changing _that_ is a much bigger task (with frontend and api work) that needs the end requirements specified in a PRD, really

This is a good point. I think when we move to using Adzerk for the primary hero, or any other location on the home page, we'll need to either:
- Just issue one query to Adzerk for the total number of creatives we need (e.g., 7 creatives - 1 for the hero and 6 for the Sponsored shelf), or
- Have separate sets of creatives for the Sponsored shelf and for the hero. If people are paying separately for each feature (shelf vs. hero) we'll need to do this, but, as Andrew pointed out, ensuring that the same add-on doesn't appear in both places will be tricky.

**Comment by eviljeff** [comment]
> This is a good point. I think when we move to using Adzerk for the primary hero, or any other location on the home page, we'll need to either:
> 
>     * Just issue one query to Adzerk for the total number of creatives we need (e.g., 7 creatives - 1 for the hero and 6 for the Sponsored shelf), or
> 
>     * Have separate sets of creatives for the Sponsored shelf and for the hero. If people are paying separately for each feature (shelf vs. hero) we'll need to do this, but, as Andrew pointed out, ensuring that the same add-on doesn't appear in both places will be tricky.

Whatever we end up doing should be driven by a PRD that specifies what the (ideal) end result on the frontend is (and then planned specifically in it's own issues).  If this issue is about adding functionality which _might_ be useful after further changes in Q1 are complete then it'll be best to table it for now, until when we know what the overall picture is.

**Comment by jvillalobos** [comment]
> are you asking that if Verified and Line add-ons are added to Adzerk they will show up in the /sponsored endpoint shelf?

That's all I'm asking. We will manually pad the list for the secondary hero (with Verified or Line) if we think that's necessary. All I'm requesting here is that AMO doesn't filter the results to strictly Sponsored add-ons.

Any work for the primary hero is meant for Q1 and later, and out of scope for this. Jumping ahead a bit, I expect there to be a separate Adzerk Flight that will deal with primary hero content, and any concerns with duplication will be dealt with operationally and not in code.

**Comment by ioanarusiczki** [comment]
@eviljeff On AMO dev:
I removed a sponsored addon [620004](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/promotedaddon/8/change/) and the [API ](https://addons-dev.allizom.org/api/v4/shelves/sponsored/)returns now 5 add-ons - all sponsored., meaning it's the same list from https://github.com/mozilla/addons-server/issues/15617#issuecomment-708289590

In Adzerk I tried adding a [line](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/promotedaddon/22/change/?_changelist_filters=group_id__exact%3D4) and a [verified ](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/promotedaddon/316/change/) addon and I was hoping the API would return one of them after making several calls. 

I don't know if my testing method is good. -> I added 620045 and 620284 to the [Adzerk flight 
](https://app.adzerk.com/#!/flight/12435798/edit/)

**Comment by eviljeff** [comment]
I've been looking this for a while and still have no clue as to why Adzerk isn't serving the new creatives.  There are 8 active creatives; 6 are the originals; one of the 6 is now not a promoted addon on AMO; there are 2 new new creatives.  

Adzerk doesn't seem to be serving them at all - if they were being served and prevented from appearing in the AMO code then the logging/stats in #15919 would highlight that - instead it's only 620004 that's logged.

**BUT - both addons added, 620045 and 620284, aren't even public on AMO.  620045 is disabled, and 620284 is awaiting review.**

**Comment by ioanarusiczki** [comment]
@eviljeff I approved 620284 and added another id in the -dev flight, it's 620290. Starting to look good 👍

**Comment by ioanarusiczki** [comment]
@eviljeff 
I don't know what's going on. 

AMO dev:

I added this morning to the Adzerk Dev Flight 2 new creatives - line addons, 619721,620292.  Right now, 6 addons are active in the Adzerk list and I removed entirely the rest of them. 

Results: 4 or 5 addons show up.

![not 6](https://user-images.githubusercontent.com/33448286/98919686-cb299280-24d7-11eb-9953-2ad6e6664db3.gif)

Active addons ids on both Adzerk and AMO: 619721,620031,620292,620290,620119,502855

AMO stage:
I left 6 add-ons "active" in the flight and added a new line addon this morning: { "id": "1005144"} and I have now the same result as on -dev: 5 addons are displayed by the frontend.
